librz/core/cmd_api.c

```diff
@@ -1089,6 +1089,92 @@ static char *get_help(RzCmd *cmd, RzCmdDesc *cd, RzCmdParsedArgs *args, bool use
 	return NULL;
 }
 
+static void fill_args_json(RzCmdDesc *cd, PJ *j) {
+	const RzCmdDescArg *arg;
+	bool has_array = false;
+	pj_k(j, "args");
+	pj_a(j);
+	const char* argtype = NULL;
+	for (arg = cd->help->args; arg && arg->name; arg++) {
+		if (has_array) {
+			rz_warn_if_reached();
+			break;
+		}
+		pj_o(j);
+#define CASE_TYPE(x,y) case (x): \
+						argtype = (y); \
+						break
+		switch(arg->type) {
+		CASE_TYPE(RZ_CMD_ARG_TYPE_FAKE, "fake");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_NUM, "number");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_RZNUM, "expression");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_STRING, "string");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_ENV, "environment_variable");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_ZIGN, "zignature");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_ZIGN_SPACE, "zignature_space");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_CHOICES, "choice");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_FCN, "function");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_FILE, "filename");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_OPTION, "option");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_CMD, "command");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_MACRO, "macro");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_EVAL_KEY, "evaluable");
+		CASE_TYPE(RZ_CMD_ARG_TYPE_EVAL_FULL, "evaluable_full");
+#undef CASE_TYPE
+		default:
+			argtype = "unknown";
+			break;
+		}
+		pj_ks(j, "type", argtype);
+		pj_ks(j, "name", arg->name);
+		if (arg->type == RZ_CMD_ARG_TYPE_FAKE) {
+			pj_end(j);
+			continue;
+		}
+		pj_kb(j, "nospace", arg->no_space);
+		pj_kb(j, "required", !arg->optional);
+		pj_kb(j, "is_array", arg->flags & RZ_CMD_ARG_FLAG_ARRAY);
+		pj_kb(j, "is_option", arg->flags & RZ_CMD_ARG_FLAG_OPTION);
+		if (arg->default_value) {
+			pj_kb(j, "default", arg->default_value);
+		}
+		pj_end(j);
```

Comment:
if arg is `choice`  add `choices` field as well.

Reply:
can you tell me an example of command with that?

Reply:
Grep in `cmd_descs.yaml` for `choices` and you'll find `sn`(seek next)  or `za`